cocotb/decorators.py

```diff
@@ -291,6 +287,7 @@ class coroutine(object):
     """
 
     def __init__(self, func):
+        func = types.coroutine(func)
```

Comment:
Does this affect stack tracing at all?
From `types.coroutine` docs:
> If gen_func is a generator function, it will be modified in-place.
> 
> If gen_func is not a generator function, it will be wrapped. If it returns an instance of collections.abc.Generator, the instance will be wrapped in an awaitable proxy object. All other types of objects will be returned as is.